### PR TITLE
add auto reconnection logic, when connection create fails.

### LINKED ISSUE #117: connection retry
### Is your proposal related to a problem?
Some time fails create connection between tape and peer.

### Describe the solution you'd like
add a general 3 times retry for connection creation before error.

### Describe alternatives you've considered
N/A

### Additional context
N/A

### PR DESCRIPTION
**Pull request by SamYuan1990**
resolve #117

Signed-off-by: SamYuan1990 <yy19902439@126.com>

### CODE REVIEW

File: pkg/infra/client.go [hunk 1]
```diff
@@ -46,43 +47,42 @@ func CreateGRPCClient(node Node) (*comm.GRPCClient, error) {
...
-func CreateDeliverFilteredClient(node Node) (peer.Deliver_DeliverFilteredClient, error) {
+func DailConnection(node Node) (*grpc.ClientConn, error) {
 	gRPCClient, err := CreateGRPCClient(node)
 	if err != nil {
 		return nil, err
 	}
-
-	conn, err := gRPCClient.NewConnection(node.Addr, func(tlsConfig *tls.Config) { tlsConfig.InsecureSkipVerify = true })
+	for i := 0; i < 3; i++ {
+		conn, err := gRPCClient.NewConnection(node.Addr, func(tlsConfig *tls.Config) { tlsConfig.InsecureSkipVerify = true })
```
Comment (guoger) [inline pkg/infra/client.go:79]: 这里的`err`是一个新的var，80行的`err`不会用这个值

**Review by guoger — CHANGES_REQUESTED** [review]

File: pkg/infra/client.go [hunk 2]
```diff
@@ -46,43 +47,43 @@ func CreateGRPCClient(node Node) (*comm.GRPCClient, error) {
...
+	var connError error
+	for i := 0; i < 3; i++ {
+		conn, connError := gRPCClient.NewConnection(node.Addr, func(tlsConfig *tls.Config) { tlsConfig.InsecureSkipVerify = true })
+		if connError == nil {
+			return conn, nil
+		}
+	}
+	if connError != nil {
+		return nil, errors.Wrapf(connError, "error connecting to %s", node.Addr)
+	}
```
Comment (guoger) [inline pkg/infra/client.go:87]: if we are here, it's certain that `connError != nil`, right?

File: pkg/infra/client.go [hunk 3]
```diff
@@ -46,43 +47,43 @@ func CreateGRPCClient(node Node) (*comm.GRPCClient, error) {
...
+	gRPCClient, err := CreateGRPCClient(node)
 	if err != nil {
-		return nil, errors.Wrapf(err, "error connecting to %s", node.Addr)
+		return nil, err
 	}
-
-	return peer.NewDeliverClient(conn).DeliverFiltered(context.Background())
+	var connError error
+	for i := 0; i < 3; i++ {
+		conn, connError := gRPCClient.NewConnection(node.Addr, func(tlsConfig *tls.Config) { tlsConfig.InsecureSkipVerify = true })
```
Comment (guoger) [inline pkg/infra/client.go:80]: this still create a new `connError` within the for scope

**Review by guoger — CHANGES_REQUESTED** [review]

File: pkg/infra/client.go (hunk 3 above)
Comment (SamYuan1990) [reply pkg/infra/client.go]: fixed for this one.

File: pkg/infra/client.go (hunk 2 above)
Comment (SamYuan1990) [reply pkg/infra/client.go:88]: right.

File: pkg/infra/client.go [hunk 4]
```diff
@@ -45,44 +47,45 @@ func CreateGRPCClient(node Node) (*comm.GRPCClient, error) {
...
-		return nil, errors.Wrapf(err, "error connecting to %s", node.Addr)
+		return nil, err
 	}
-
-	return orderer.NewAtomicBroadcastClient(conn).Broadcast(context.Background())
+	return peer.NewDeliverClient(conn).DeliverFiltered(context.Background())
 }
 
-func CreateDeliverFilteredClient(node Node) (peer.Deliver_DeliverFilteredClient, error) {
+// TODO: use a global get logger function instead inject a logger
```
Comment (guoger) [inline pkg/infra/client.go:74]: why? isn't logger injection better than global getter? so you could add some customizable tags, etc

**Review by guoger — APPROVED** [review]

**Pull request merged**